avm/res/document-db/database-account/main.bicep

```
@@ -614,3 +693,54 @@ type sqlDatabaseContainer = {
     paths: string[]
   }[]?
 }
+
+type secretsKeyVault = {
+  @description('Required. The key vault name where to store the keys and connection strings generated by the modules.')
+  keyVaultName: string
+
+  @description('Optional. Default to the resource group where this account is. The resource group name where the key vault is.')
+  resourceGroupName: string?
+
+  @description('Optional. Default to Primary-Write-Key. The primary write key secret name to create.')
+  primaryWriteKeySecretName: string?
+
+  @description('Optional. Default to Primary-Readonly-Key. The primary readonly key secret name to create.')
+  primaryReadOnlyKeySecretName: string?
+
+  @description('Optional. Default to Primary-Write-ConnectionString. The primary write connection string secret name to create.')
+  primaryWriteConnectionStringSecretName: string?
+
+  @description('Optional. Default to Primary-Readonly-ConnectionString. The primary readonly connection string secret name to create.')
+  primaryReadonlyConnectionStringSecretName: string?
+
+  @description('Optional. Default to Secondary-Write-Key. The primary write key secret name to create.')
+  secondaryWriteKeySecretName: string?
+
+  @description('Optional. Default to Secondary-Readonly-Key. The primary readonly key secret name to create.')
+  secondaryReadonlyKeySecretName: string?
+
+  @description('Optional. Default to Secondary-Write-ConnectionString. The primary write connection string secret name to create.')
+  secondaryWriteConnectionStringSecretName: string?
+
+  @description('Optional. Default to Secondary-Readonly-ConnectionString. The primary readonly connection string secret name to create.')
+  secondaryReadonlyConnectionStringSecretName: string?
+}
+
+type networkRestrictionsConfig = {
+  @description('Optional. Default to []. List of Ip or CIDR ranges.')
+  ipRules: string[]
+
+  @description('Optional. Default to AzureServices. Specifies the network ACL bypass for Azure services.')
+  networkAclBypass: ('AzureServices' | 'None')?
+
+  @description('Optional. Default to Enabled. Whether requests from Public Network are allowed.')
+  publicNetworkAccess: ('Enabled' | 'Disabled')?
+
+  @description('Optional. Default to []. List of Virtual Network ACL rules configured for the Cosmos DB account..')
+  virtualNetworkRules: networkVnetRestrictionsConfig[]
+}
+
+type networkVnetRestrictionsConfig = {
+  @description('Required. A single IPv4 address or a single IPv4 address range in CIDR format. Provided IPs must be well-formatted and cannot be contained in one of the following ranges: 10.0.0.0/8, 100.64.0.0/10, 172.16.0.0/12, 192.168.0.0/16, since these are not enforceable by the IP address filter. Example of valid inputs: “23.40.210.245” or “23.40.210.0/8”.')
```

```suggestion
  @description('Required. A single IPv4 address or a single IPv4 address range in CIDR format. Provided IPs must be well-formatted and cannot be contained in one of the following ranges: 10.0.0.0/8, 100.64.0.0/10, 172.16.0.0/12, 192.168.0.0/16, since these are not enforceable by the IP address filter. Example of valid inputs: "23.40.210.245" or "23.40.210.0/8".')
```

Careful with special characters.

---

Fixed